unified_address.py

```diff
@@ -4,17 +4,18 @@
 import math
 import struct
 
-from pyblake2 import blake2b
 from bech32m import bech32_encode, bech32_decode, convertbits, Encoding
 
 from tv_output import render_args, render_tv, Some
 from tv_rand import Rand
+from zc_utils import write_compact_size, parse_compact_size
 from f4jumble import f4jumble, f4jumble_inv
 import sapling_key_components
 import orchard_key_components
 
+
 def tlv(typecode, value):
-    return b"".join([bytes([typecode, len(value)]), value])
+    return b"".join([write_compact_size(typecode), write_compact_size(len(value)), value])
```

Comment:
Similarly, we should be asserting inside `write_compact_size` that these are within the correct CompactSize range (either explicitly, or by default).